keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,903 @@
+    |       name: bucketClaim123                      |
+    |       namespace: ns1                            |
+    |   protocols:                                    |
+    |   - s3                                          |
+    |   parameters:                                   |
+    |     key: value                                  |
+    |   driverName: s3.amazonaws.com                  |
+    |-------------------------------------------------|
 ```
 
-## ProvisionerGetInfo
 
-This call is meant to retrieve the unique provisioner Identity. This identity will have to be set in `BucketRequest.Provisioner` field in order to invoke this specific provisioner.
+###### 2. User creates BucketClaim referring to the bucket
+
+Only the specified BucketClaim will be able to bind to the Bucket.
 
 ```
-message ProvisionerGetInfoRequest {
-    // Intentionally left blank
-}
+    BucketClaim - bucketClaim123
+    |------------------------------------------|
+    | name: bucketClaim123                     |
+    | spec:                                    |
+    |   existingBucketName: bucketName123      |
+    |                                          |
+    |------------------------------------------|
 
-message ProvisionerGetInfoResponse {
-  // The name MUST follow domain name notation format
-  // (https://tools.ietf.org/html/rfc1035#section-2.3.1). It SHOULD
-  // include the plugin's host company name and the plugin name,
-  // to minimize the possibility of collisions. It MUST be 63
-  // characters or less, beginning and ending with an alphanumeric
-  // character ([a-z0-9A-Z]) with dashes (-), dots (.), and
-  // alphanumerics between. This field is REQUIRED.
-  string name = 1;
-}
 ```
 
-## ProvisonerCreateBucket
+###### 3. User creates BucketAccess to generate credentials for that bucket
 
-This call is made to create the bucket in the backend. If a bucket that matches both name and parameters already exists, then OK (success) must be returned. If a bucket by same name, but different parameters is provided, then the appropriate error code `ALREADY_EXISTS` must be returned by the provisioner. The call to *ProvisonerCreateBucket* MUST be idempotent.
+Similar to the BucketAccess for COSI created bucket, this BucketAccess should reference the BucketClaim that refers to this bucket.
 
 ```
-message ProvisionerCreateBucketRequest {    
-    // This field is REQUIRED
-    string bucket_name = 1;
+    BucketAccess - bac2
+    |-------------------------------|
+    | spec:                         |
+    |   bucketClaim: bucketClaim123 |
+    |                               |
+    |-------------------------------|
+```
 
-    map<string,string> bucket_context = 2;
+Note that, as of the alpha version of COSI, there is no authorization mechanism to restrict users who can refer to buckets imported into COSI. In the future, access to imported buckets will also follow the namespace selector approach described [above](#sharing-buckets).
```

Comment:
Now that the admin needs to specify the claim, I think it does achieve the ability to restrict which existing buckets users can reference.